search/includes/classes/class-search.php

```diff
@@ -237,4 +240,42 @@ static function ep_skip_query_integration( $skip ) {
 		// The filter is checking if we should _skip_ query integration
 		return ! ( $query_integration_enabled || $query_integration_enabled_legacy );
 	}
+
+	/**
+	 * Filter for ep_pre_request_host
+	 *
+	 * Return the next host in our enpoint list if it's defined. Otherwise, return the last host.
+	 */
+	public function filter__ep_pre_request_host( $host, $failures, $path, $args ) {
+		if ( ! defined( 'VIP_ELASTICSEARCH_ENDPOINTS' ) ) { 
+			return $host;
+		}
+
+		return $this->get_next_host( $host, VIP_ELASTICSEARCH_ENDPOINTS );
```

Comment:
I just realized, this will end up trying the "next" host for every request, because this filter always runs before any request. What that means is that if one of the hosts is down, it will keep being retried every Nth time (if there are enough ES requests) instead of it settling on a working host after one failed request.

Reply:
I think the ideal solution is that it uses the first one from `EP_HOST` by default (which is itself randomized)...then if and only if the request fails, it moves on to the next host in the list - otherwise it keeps hitting the host that it knows is working.